environments/gravity.py

```diff
@@ -0,0 +1,213 @@
+            raise NotImplementedError(
+                'Gravity interaction for ' + str(self.n_objects) + ' bodies is not implemented.')
+        super().__init__(q=q, p=p)
+
+    def set(self, q, p):
+        """Sets initial conditions for pendulum
+
+        Args:
+            q (ndarray): Object generalized positions in 2-D space: Positions (m)
+            p (ndarray): Object generalized momentums in 2-D space : Linear momentums (kg*m/s)
+
+        Raises:
+            ValueError: If q and p are not in 2-D space or do not refer to all the objects in space
+        """
+        if q is None or p is None:
+            return
+        if q.shape[0] != self.n_objects or p.shape[0] != self.n_objects:
+            raise ValueError(
+                "q and p do not refer to the same number of objects in the system.")
+        if q.shape[-1] != 2 or p.shape[-1] != 2:
+            raise ValueError(
+                "q and p must be in 2-D space: Position and Linear momentum.")
+        self.q = q.copy()
+        self.p = p.copy()
+
+    def _dynamics(self, t, states):
+        """Defines system dynamics
+
+        Args:
+            t (float): Time parameter of the dynamic equations.
+            states ([float]) Phase states at time t
+
+        Returns:
+            equations ([float]): Movement equations of the physical system
+        """
+        # Convert states to n_object arrays of q and p
+        states_resh = states.reshape(2, self.n_objects, 2)
+        dyn = np.zeros_like(states_resh)
+        states_q = states_resh[0, :, :]
+        states_p = states_resh[1, :, :]
+        dyn[0, :, :] = states_p/(np.array(self.mass)[:, np.newaxis])
+
+        # Distance calculation
+        object_distance = np.zeros((self.n_objects, self.n_objects))
+        for i in range(self.n_objects):
+            for j in range(i, self.n_objects):
+                object_distance[i, j] = np.linalg.norm(
+                    states_q[i] - states_q[j])
+                object_distance[j, i] = object_distance[i, j]
+        object_distance = np.power(object_distance, 3)/self.gravity_cst
+
+        for d in range(2):
+            for i in range(self.n_objects):
+                mom_term = 0
+                for j in range(self.n_objects):
+                    if i != j:
+                        mom_term += self.mass[j]*(states_q[j, d] -
+                                                  states_q[i, d])/object_distance[i, j]
+                dyn[1, i, d] += mom_term*self.mass[i]
+        return dyn.reshape(-1)
+
+    def _draw(self, res=32, color=True, world_size=1.5):
+        """Returns array of the environment evolution
+
+        Args:
+            res (int): Image resolution (images are square).
+            color (bool): True if RGB, false if grayscale.
+            world_size (float) Spatial extent of the window where the rendering is taking place (in meters).
+
+        Returns:
+            vid (np.ndarray): Rendered rollout as a sequence of images
```

Comment:
Just to make it clear the shape of returned things.
```suggestion
            vid (np.ndarray): Numpy array of shape (seq_len, height, width, channels)
                containing the rendered rollout as a sequence of images.
```